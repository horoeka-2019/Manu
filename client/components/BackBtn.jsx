import React from 'react'
import { Button } from 'semantic-ui-react'

const BackBtn = () => <Button size='big' className='backBtnStyle'>Back</Button>

export default BackBtn
