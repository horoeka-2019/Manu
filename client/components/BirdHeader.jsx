import React from 'react'
import { Menu, Container } from 'semantic-ui-react'

class BirdHeader extends React.Component {
  render () {
    return (

      <Menu className='centerHeader' borderless fixed='top' inverted style = {{ backgroundColor: '#F3A712' }}>
        <Container>
          <Menu.Item>
            He Pokai Manu
          </Menu.Item>
        </Container>
      </Menu>

    )
  }
}

export default BirdHeader
