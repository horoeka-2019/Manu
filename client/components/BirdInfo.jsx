import React from 'react'
import { Link, HashRouter as Router } from 'react-router-dom'
import BirdHeader from './BirdHeader'
import BackBtn from './BackBtn'
import BirdInfoText from './BirdInfoText'

class BirdInfo extends React.Component {
  render () {
    return (
      <React.Fragment>
        <BirdHeader />
        <BirdInfoText birds={this.props.birds} {...props}/>
        {/* <Link to={`/profile/${props.match.params.id}`}>
          <BackBtn/>
        </Link> */}
      </React.Fragment>
    )
  }
}

export default BirdInfo
