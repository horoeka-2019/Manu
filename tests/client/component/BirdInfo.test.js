import React from 'react'
import { mount } from 'enzyme'
import BirdInfo from '../../../client/components/BirdInfo'
import BirdHeader from '../../../client/components/BirdHeader'
import BackBtn from '../../../client/components/BackBtn'
import { HashRouter as Router } from 'react-router-dom'

const mockBirdsArray = [{
  bird_id: 1,
  found: 1,
  image: '/images/tui.jpg',
  info: 'Tui are unique to New Zealand and belong to the honey-eater family which means that they feed mainly on nectar from the flowers of native plants. They can be quite aggressive! They will chase other Tui away from food and tasty nectar. Tui are found throughout the North, South and Stewart Islands of New Zealand.',
  name: 'Tui'
}]

describe('BirdInfo component test', () => {
  it('BirdInfo contains header, info card, and button components', () => {
    // Arrange
    const params = { params: { id: 1 } }
    const expected = [<BirdHeader/>, <BackBtn/>]
    // Act
    const wrapper = mount(<Router><BirdInfo match={params} birds={mockBirdsArray}/></Router>)
    const actual = wrapper.containsAllMatchingElements(expected)
    // Assert
    expect(actual).toBeTruthy()
  })
  it('BirdInfo Card element has expected props', () => {
    // Arrange
    const params = { params: { id: 1 } }
    const { name, info } = mockBirdsArray[0]

    // Act
    const wrapper = mount(<Router><BirdInfo match={params} birds={mockBirdsArray}/></Router>)
    const actual = wrapper.setProps({ name: name, info: info })
    const actualInfo = wrapper.text()

    // Assert
    expect(actual).hasOwnProperty('name')
    expect(actual).hasOwnProperty('info')
    expect(actualInfo).toContain(info)
  })
})
