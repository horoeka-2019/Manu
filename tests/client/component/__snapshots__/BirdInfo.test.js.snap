// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BirdInfo component test render unchanged from previous snapshot 1`] = `
<HashRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <BirdInfo
      birds={
        Array [
          Object {
            "bird_id": 1,
            "found": 1,
            "image": "/images/tui.jpg",
            "info": "Tui are unique to New Zealand and belong to the honey-eater family which means that they feed mainly on nectar from the flowers of native plants. They can be quite aggressive! They will chase other Tui away from food and tasty nectar. Tui are found throughout the North, South and Stewart Islands of New Zealand.",
            "name": "Tui",
          },
        ]
      }
      match={
        Object {
          "params": Object {
            "id": "1",
          },
        }
      }
    >
      <div
        className="birdInfoWrapper"
      >
        <BirdHeader>
          <Menu
            borderless={true}
            className="centerHeader"
            fixed="top"
            inverted={true}
            style={
              Object {
                "backgroundColor": "#F3A712",
              }
            }
          >
            <div
              className="ui borderless inverted top fixed centerHeader menu"
              style={
                Object {
                  "backgroundColor": "#F3A712",
                }
              }
            >
              <MenuItem>
                <div
                  className="item"
                  onClick={[Function]}
                >
                  He Pokai Manu
                </div>
              </MenuItem>
            </div>
          </Menu>
        </BirdHeader>
        <BirdInfoText
          birds={
            Array [
              Object {
                "bird_id": 1,
                "found": 1,
                "image": "/images/tui.jpg",
                "info": "Tui are unique to New Zealand and belong to the honey-eater family which means that they feed mainly on nectar from the flowers of native plants. They can be quite aggressive! They will chase other Tui away from food and tasty nectar. Tui are found throughout the North, South and Stewart Islands of New Zealand.",
                "name": "Tui",
              },
            ]
          }
          match={
            Object {
              "params": Object {
                "id": "1",
              },
            }
          }
        >
          <ScrollToTop />
          <div
            className="birdInfoTextWrapper"
          >
            <div
              className="birdInfoTextName"
            >
              Tui
            </div>
            <div
              className="birdInfoTextInfo"
            >
              Tui are unique to New Zealand and belong to the honey-eater family which means that they feed mainly on nectar from the flowers of native plants. They can be quite aggressive! They will chase other Tui away from food and tasty nectar. Tui are found throughout the North, South and Stewart Islands of New Zealand.
            </div>
            <br />
            <div
              className="birdInfoTextInfo"
            >
              Conservation Status:
            </div>
            <div
              className="threatened"
            />
          </div>
        </BirdInfoText>
        <Link
          to="/profile/1"
        >
          <LinkAnchor
            href="#/profile/1"
            navigate={[Function]}
          >
            <a
              href="#/profile/1"
              onClick={[Function]}
            >
              <BackBtn
                className="backBtnStyle"
              >
                <Button
                  as="button"
                  className="backBtnStyle"
                  size="big"
                >
                  <Ref
                    innerRef={
                      Object {
                        "current": <button
                          class="ui big button backBtnStyle"
                        >
                          Back
                        </button>,
                      }
                    }
                  >
                    <RefFindNode
                      innerRef={
                        Object {
                          "current": <button
                            class="ui big button backBtnStyle"
                          >
                            Back
                          </button>,
                        }
                      }
                    >
                      <button
                        className="ui big button backBtnStyle"
                        onClick={[Function]}
                      >
                        Back
                      </button>
                    </RefFindNode>
                  </Ref>
                </Button>
              </BackBtn>
            </a>
          </LinkAnchor>
        </Link>
      </div>
    </BirdInfo>
  </Router>
</HashRouter>
`;
